### PR TITLE
Convert some tools from deprecated OptionParser to ArgParser

### PR DESCRIPTION
**Pull request by pinkavaj**
Other will follow in future

### CODE REVIEW

**Comment by jmcorgan** [comment]
@skoslowski Can you look at the grc commits?  I'll merge from here if that's ok.

**Comment by mbr0wn** [comment]
Yay ArgParse! We need to make sure we don't break our minimum requirements, though. Current GR branch officially supports Python 2.5, I don't think that includes ArgParser.

**Comment by pinkavaj** [comment]
ArgParser is from 2.7. Is there intention to drop support for Python 2.5/2.6 in near future? request is for next exactly for this reason.

**Comment by mbr0wn** [comment]
next will bump the Python versions, I mainly wanted to point this out to @jmcorgan because we haven't finalized the new version requirements yet.

**Comment by trondeau** [comment]
@mbr0wn we haven't finalized all of the version bumps for 3.8, yet, but I imagine Python 2.7 is going to make the cut.

**Comment by jmcorgan** [comment]
Yes, this will get merged onto next soon.  We'll definitely have 2.7.x still for next, but which 'x' is not yet determined.

**Comment by jmcorgan** [comment]
@skoslowski Can you look at the grc changes?

**Comment by skoslowski** [comment]
will do

**Comment by jmcorgan** [comment]
@skoslowski would like to move on this soon

File: grc/scripts/gnuradio-companion [hunk 1]
```diff
@@ -98,10 +98,10 @@ def main():
     check_gnuradio_import()
 
     from gnuradio import gr
-    parser = optparse.OptionParser(
-        usage='usage: %prog [options] [saved flow graphs]',
-        version=VERSION_AND_DISCLAIMER_TEMPLATE % gr.version())
-    options, args = parser.parse_args()
+    parser = argparse.ArgumentParser(
+        description=VERSION_AND_DISCLAIMER_TEMPLATE % gr.version())
+    parser.add_argument('flow_graphs')
```
Comment (skoslowski) [inline grc/scripts/gnuradio-companion]: This should be 
```python
parser.add_argument('flow_graphs', nargs='*')
```

File: grc/python/flow_graph.tmpl [hunk 2]
```diff
@@ -333,13 +333,13 @@ $short_id#slurp
...
-    parser = OptionParser(option_class=eng_option, usage="%prog: [options]")
+    parser = ArgumentParser()
     #for $param in $parameters
         #set $type = $param.get_param('type').get_value()
         #if $type
             #silent $params_eq_list.append('%s=options.%s'%($param.get_id(), $param.get_id()))
-    parser.add_option(
-        "$make_short_id($param)", "--$param.get_id().replace('_', '-')", dest="$param.get_id()", type="$type", default=$make_default($type, $param),
+    parser.add_argument(
+        "$make_short_id($param)", "--$param.get_id().replace('_', '-')", dest="$param.get_id()", type=$type, default=$make_default($type, $param),
```
Comment (skoslowski) [inline grc/python/flow_graph.tmpl]: not sure this is (or ever was) safe:
```
type=$type
```
From parameter.xml I see values like eng_float or intx for which we don't have handlers here.
I recommend adding a new <opt> in the xml to specify the argparse_type. Read it by changing line 338 to ``` $param.get_param('type').get_option('argparse_type')```

**Comment by jmcorgan** [comment]
@pinkavaj Do you plan to address the comments from Sebastian?

**Comment by pinkavaj** [comment]
Yes, but I'm now a bit busy, try to do it in a week or two.

File: grc/python/flow_graph.tmpl (hunk 2 above)
Comment (pinkavaj) [inline grc/python/flow_graph.tmpl]: I'm not sure I understand, do You mean add into grc/blocks/options.xml something like:

        

?

Comment (skoslowski) [inline grc/python/flow_graph.tmpl]: Yes, that works for me

**Comment by pinkavaj** [comment]
I did git push -f, those commits should confirm with the review comments.

**Comment by jmcorgan** [comment]
@skoslowski please confirm mergeability

**Comment by jmcorgan** [comment]
With the recent GRC refactoring, this branch has merge conflicts.

@pinkavaj Can you redo this branch off current next branch and force push?

**Comment by pinkavaj** [comment]
Fixed collision in gnuradio-companion

File: grc/core/generator/flow_graph.tmpl [hunk 3]
```diff
@@ -336,18 +336,18 @@ $short_id#slurp
...
 def argument_parser():
-    #set $desc_args = 'usage="%prog: [options]", option_class=eng_option'
     #if $flow_graph.get_option('description')
-        #set $desc_args += ', description=description'
     description = $repr($flow_graph.get_option('description'))
+    #else
+    description = None
     #end if
-    parser = OptionParser($desc_args)
+    parser = ArgumentParser(description=description)
```
Comment (skoslowski) [inline grc/core/generator/flow_graph.tmpl]: Please don't change the structure here. We only want to emit arguments we really need. That is why the arguments to ArgumentParser() are first collected in a string.

File: grc/main.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
-import optparse
+from argparse import ArgumentParser
```
Comment (skoslowski) [inline grc/main.py]: (minor) --> "import argparse"

**Comment by skoslowski** [comment]
I put two comments inline.
The rest is good to go.

**Comment by pinkavaj** [comment]
Should be ok now, thanks for comments.

**Comment by skoslowski** [comment]
Looks good. thanks.

**Pull request merged**